manubot/process/util.py

```diff
@@ -285,51 +307,50 @@ def get_citekeys_df(args, text):
             tag_df = pandas.read_csv(args.citation_tags_path, delim_whitespace=True)
         tag_df["manuscript_citekey"] = "tag:" + tag_df.tag
         tag_df = tag_df.rename(columns={"citation": "detagged_citekey"})
-        for detagged_citekey in tag_df.detagged_citekey:
-            is_valid_citekey(detagged_citekey, allow_raw=True)
-        citekeys_df = citekeys_df.merge(
-            tag_df[["manuscript_citekey", "detagged_citekey"]], how="left"
+        citekey_aliases = dict(
+            zip(tag_df["manuscript_citekey"], tag_df["detagged_citekey"])
         )
     else:
-        citekeys_df["detagged_citekey"] = None
+        citekey_aliases = {}
         logging.info(
             f"missing {args.citation_tags_path} file: no citation tags (citekey aliases) set"
         )
-    citekeys_df.detagged_citekey.fillna(
-        citekeys_df.manuscript_citekey.astype(str), inplace=True
-    )
-    citekeys_df["standard_citekey"] = citekeys_df.detagged_citekey.map(
-        standardize_citekey
-    )
-    citekeys_df["short_citekey"] = citekeys_df.standard_citekey.map(shorten_citekey)
-    citekeys_df = citekeys_df.sort_values(["standard_citekey", "detagged_citekey"])
+    citekeys_df = get_citekeys_df(manuscript_citekeys, citekey_aliases)
     citekeys_df.to_csv(args.citations_path, sep="\t", index=False)
-    check_collisions(citekeys_df)
-    check_multiple_citation_strings(citekeys_df)
     return citekeys_df
 
 
-def generate_csl_items(args, citekeys_df):
+def generate_csl_items(
+    citekeys, manual_refs={}, requests_cache_path=None, clear_requests_cache=False
+):
     """
     General CSL (citeproc) items for standard_citekeys in citekeys_df.
-    Writes references.json to disk and logs warnings for potential problems.
-    """
-    # Read manual references (overrides) in JSON CSL
-    manual_refs = load_manual_references(args.manual_references_paths)
 
-    requests  # require `import requests` in case this is essential for monkey patching by requests_cache.
-    requests_cache.install_cache(args.requests_cache_path, include_get_headers=True)
-    cache = requests_cache.get_cache()
-    if args.clear_requests_cache:
-        logging.info("Clearing requests-cache")
-        requests_cache.clear()
-    logging.info(
-        f"requests-cache starting with {len(cache.responses)} cached responses"
-    )
+    Parameters
```

Comment:
Could list the other two parameters here: `requests_cache_path` and `clear_requests_cache`

Reply:
done in https://github.com/manubot/manubot/pull/190/commits/0d1c6d113f23db142b8dcb8af805613cc212e1a1